pywbem/_logging.py

```diff
@@ -14,404 +14,341 @@
-  logger.
-
-These functions save the logger definitions in the
-:class:`~pywbem._logging.PywbemLoggers` class that
-is used by the logging functions in the recorder so trying to create the named
-loggers independently of this code may cause issues.
-
-The pywbem loggers are based on two parameters (log destination, log detail)
-that determine if the logs for the logger name are created,
-how much information is inserted, and the log destination.  This
-extends the python logging facility to include the log_detail parameter which
-defines whether all the information defined or only a limited size is output.
-This is used with pywbem because the logs on operation and http responses can be
-very large.
-
-The code that executes the loggers call the function get_logger(..) to get a
-logger from a defined logger name.  If that logger has not yet been defined
-in PywbemLoggers, an entry will be added with the default parameters.
-"""
-from __future__ import absolute_import
+Pywbem logging is used to record information passing between the pywbem client
+and WBEM servers but not as a general recorder for errors, state, etc. within
+pywbem. In effect it is a trace tool. Pywbem errors are generally passed to
+the pywbem API user as Python exceptions rather than being recorded in a log
+by a pywbem logger.
+
+Pywbem logging defines two Python :class:`~py:logging.Logger` objects
+(named loggers):
+
+* ``pywbem.api`` - Logs user-issued calls to :class:`~pywbem.WBEMConnection`
+  methods that drive WBEM operations (see :ref:`WBEM operations`) and their
+  responses before they are passed back to the user. Log entries are defined at
+  DEBUG log level. This logs the parameters of each request and the results
+  including CIMObjects/exceptions in each method response. It also logs the
+  creation of the :class:`~pywbem.WBEMConnection` object to capture connection
+  information.
+
+* ``pywbem.http`` - Logs HTTP requests and responses between the pywbem client
+  and WBEM server. This logs the http request data and response data including
+  http headers and XML payload.
+
+The above logger names output log records when they are configured and
+activated. All pywbem loggers  log at the DEBUG log level and
+logging is activated for one or more :class:`~pywbem.WBEMConnection` instance
+objects.
+
+There are three steps to configuration of the pywbem loggers:
+
+* **Configure Python logger names** - Sets the Python logger parameters
+  (logging handlers, message formats, etc.). Either Python logging
+  methods or methods defined in :ref:`Logging configuration methods` may be
+  used to configure the loggers. The `dest` parameter of the pywbem methods
+  configures the loggers for either stream or file loggers and the `name`
+  parameter defines which of the pywbem loggers is being configured. This
+  configure step does not activate any pywbem logger. If no loggers are
+  configured, pywbem defaults to a logger with the Python
```

Comment:
On "If no loggers are configured, pywbem defaults to a logger ...": That is confusing to me. It seems to imply that there are different loggers in place dependent on whether or not I configure them. That is not the case, it is always the same set of two loggers.

Suggest to change this to:

"By default, the two pywbem loggers have null handlers (see logging.NullHandler) configured so that logging is not propagated up the Python logger hierarchy."

Reply:
DONE